mathjax3-ts/input/tex/TexParser.ts

```diff
@@ -171,135 +164,311 @@ export default class TexParser {
+   * @param {StackItem|MmlNode[]} args The new items.
+   */
+  public PushAll(args: (StackItem|MmlNode)[]) {
+    for (let i = 0, m = args.length; i < m; i++) {
+      this.stack.Push(args[i]);
+    }
+  }
+
 
   /**
-   * @constructor
-   * @param {Array.<string>} maps Names of the maps included in this
-   *     configuration.
+   * @return {MmlNode} The internal Mathml structure.
    */
-  constructor(maps: string[]) {
-    for (const name of maps) {
-      this.add(name);
+  public mml(): MmlNode {
+    if (!this.stack.Top().isKind('mml')) {
+      return null;
     }
+    let node = this.stack.Top().Top;
+    this.configuration.popParser();
+    return node;
   }
 
+  /************************************************************************
+   *
+   *   String handling routines
+   */
 
   /**
-   * Sets the default method to call when parsing fails.
-   * @param {function(string): ParseResult} method The fallback method.
+   *  Convert delimiter to character
    */
-  public fallback(method: ParseMethod) {
-    this._fallback = method;
+  public convertDelimiter(c: string): string {
+    const symbol = this.lookup('delimiter', c) as Symbol;
+    return symbol ? symbol.char : null;
+  }
+
+  /**
+   *   Check if the next character is a space
+   */
+  public nextIsSpace() {
+    return this.string.charAt(this.i).match(/\s/);
   }
 
 
   /**
-   * Adds a symbol map to the configuration if it exists.
-   * @param {string} name of the symbol map.
+   *  Get the next non-space character
    */
-  public add(name: string): void {
-    let map = MapHandler.getInstance().getMap(name);
-    if (!map) {
-      this.warn('Configuration ' + name + ' not found! Omitted.');
-      return;
+  public GetNext(): string {
+    while (this.nextIsSpace()) {
+      this.i++;
     }
-    this._configuration.push(map);
+    return this.string.charAt(this.i);
   }
 
+  /**
+   *  Get and return a control-sequence name
+   */
+  public GetCS() {
+    let CS = this.string.slice(this.i).match(/^([a-z]+|.) ?/i);
+    if (CS) {
+      this.i += CS[1].length;
+      return CS[1];
+    } else {
+      this.i++;
+      return ' ';
+    }
+  }
 
   /**
-   * Parses the given input with the first applicable symbol map.
-   * @param {ParseInput} input The input for the parser.
-   * @return {ParseResult} The output of the parsing function.
+   *  Get and return a TeX argument (either a single character or control sequence,
+   *  or the contents of the next set of braces).
    */
-  public parse(input: ParseInput): ParseResult {
-    // TODO: Can't be done with applicable due to delimiter parsing!
-    for (let map of this._configuration) {
-      const result = map.parse(input);
-      if (result) {
-        return result;
+  public GetArgument(name: string, noneOK?: boolean) {
+    switch (this.GetNext()) {
+    case '':
+      if (!noneOK) {
+        // @test MissingArgFor
+        throw new TexError('MissingArgFor', 'Missing argument for %1', this.currentCS);
```

Comment:
Unfortunately, that does not longer work with the new implementation of the NewCommand package. (In particular the treatment of let.) The use of currentCS is necessary to get meaningful error messages. Otherwise you get the name of the last macro being expanded instead of the calling macro.